constructs/lambda-service.ts

```diff
@@ -17,19 +17,47 @@ import { ServiceQueueFunction } from './service-queue-function';
 import { ServiceCronFunction } from './service-cron-function';
 import { ServiceEventFunction } from './service-event-function';
 import { validatePathParameters } from '../util/validate-path-parameters';
+import {
+	ApiGateway,
+	ApiStage,
+	isLambdaAuthorizerConfig,
+	JwtAuthorizer,
+	JwtAuthorizerConfig,
+	LambdaAuthorizer,
+	LambdaAuthorizerConfig,
+} from './api-gateway';
+import { CfnAuthorizer } from 'aws-cdk-lib/aws-apigatewayv2';
 
 export interface LambdaServiceProps {
-	handlersFolder: string;
-	jwtAuthorizer?: {
-		identitySource: string[];
-		audience: string[];
-		issuer: string;
-	};
-	lambdaAuthorizer?: {
-		fn: lambda.IFunction;
-		identitySource: string[];
-		enableSimpleResponses?: boolean;
-	};
+	/** The path to the folder where the handlers are stored.
+	 *
+	 * If omitted, then the service will have no handlers. This can be useful if
+	 * you want to create a dummy service that will contain all the resources you
+	 * may need like an api gateway, which you can then pass into other actual
+	 * services.
+	 */
+	handlersFolder?: string;
+	/** The API gateway that the API handlers in this service should be attached
+	 * to.
+	 *
+	 * If this is not provided and the service includes API handlers, a new API
+	 * gateway will be created.
+	 */
+	api?: apigwv2.CfnApi;
+	/** The API gateway stage that the API handlers in this service should be
+	 * attached to.
+	 *
+	 * If this is not provided and the service includes API handlers, a new API
+	 * gateway stage will be created.
+	 */
+	stage?: apigwv2.CfnStage;
+	/** The Authorizer to use for the API handlers.
+	 *
+	 * This can either be an actual authorizer, in which case we'll use it. Or it
+	 * can be a configuration for either a Jwt or Lambda authorizer, in which case
+	 * we'll create a new authorizer with that configuration.
+	 */
+	authorizer?: JwtAuthorizerConfig | LambdaAuthorizerConfig | CfnAuthorizer;
```

Comment:
🍟 Just for those who are unaware of this PR, we should keep `lambdaAuthorizer` and `jwtAuthorizer` but set their typing to `never` and have a `@deprecated` comment directing them to `authorizer`.

Reply:
Great feedback, thanks @s-fletcher